otelcol/collector.go

```diff
@@ -105,8 +112,20 @@ type Collector struct {
 
 // NewCollector creates and returns a new instance of Collector.
 func NewCollector(set CollectorSettings) (*Collector, error) {
-	if set.ConfigProvider == nil {
-		return nil, errors.New("invalid nil config provider")
+	var err error
+	configProvider := set.ConfigProvider
+
+	if configProvider == nil {
+		resolverSet := set.ConfigProviderSettings.ResolverSettings
+		// Provide a default set of providers and converters if none have been specified.
+		if len(resolverSet.Providers) == 0 && len(resolverSet.Converters) == 0 {
+			set.ConfigProviderSettings = newDefaultConfigProviderSettings(resolverSet.URIs)
+		}
```

Comment:
Should we return an error here instead and expose `newDefaultConfigProviderSettings` as part of the public API?

Reply:
Good callout. I would like to keep `newDefaultConfigProviderSettings` private and the defaults list to the builder once this change is in. Ideally we could remove the default providers and converters from the dependency tree for Collector builds that don't use them.

This statement was to make it so this PR isn't a breaking change, but I think it's better placed in `NewCommand` so we don't have to make a breaking change in `NewCollector`, so I moved it there.

Reply:
Makes sense, thanks!